### DISCUSSION TITLE
New Feature Ideas

### DISCUSSION THREAD

**Discussion by Thinkersbluff** [Thinkersbluff/DGUS-Reloaded_for_CR6-DWIN-SET_Component#8] [Ideas]
I hope to put out small incremental improvements to this UI, rather than save things up for periodic major release cycles.  
Think "Agile" product development life-cycle, incremental build, incremental release. 

My challenge is to learn just enough to be able to implement each feature "wisely".  (i.e in a manner that is consistent with the software architecture as it is and/or as it should be, for future robustness and maintenance.)

e.g. Adding a button that just calls a gcode macro (like M112) is easy for me now.  I am thinking M600 might be handy, on the Change Filament or Printing Tune menu.

To add an extruder esteps calibration function, I would require a bit of help on the python side.  I would also expect to be implementing a rotational-distance calibration tool, rather than esteps.  Maybe the first release would require manual editing of the printer.cfg file, while upgrades could read/write the values?

Think "Agile" product development life-cycle, incremental build, incremental release.

**Comment by henriquevianasilva** [comment]
I want to say that you are doing a great job and thanks to it we are giving our lcd a second life and improving the way we use klipper.


I have three ideas I'd like to share.

1. Turn on/off the hotend led light.

2. Keep klipper up to date, dgus version is 4/5 months old.

3. Would it be possible for you to be able to adapt the second dgus approach?

[https://github.com/Desuuuu/klipper/discussions/74](https://github.com/Desuuuu/klipper/discussions/74)


In the next few days I will test your last version well and give you new feedback.

**Reply by Thinkersbluff** [reply]
Thank you for the encouragement!

1. The Hotend LED on/off is presently implemented only at the bottom of the print Tune menu. Not visible until you are printing. 

For which additional task(s) will you find it helpful? It would not take much effort for me to do add it to another of the menus. 

2. I recognize the problem with Klipper going out of date. This is a fork of a fork. If Desuuuu is not keeping his end fresh, neither will this be updated.  I will need help from folks with GitHub and Klipper, to start updating this version from the Master branch again, if that can be done. It is on the medium to longer term to-do list to solve this issue. 

3. No, I do not plan to adapt his dynamic display solution. I am sticking with the static display.  Desuuuu himself will not commit to a schedule for finishing his dynamic display, for now.

**Reply by henriquevianasilva** [reply]
I still haven't had time to test the print in this new version, so I didn't see the LED option!
Thanks again .


To help you with the update issue, the version of gbkwiatt seems to be more up-to-date and compatible with the original fork. https://github.com/gbkwiatt/klipper/tree/dgus-reloaded


Other ideas to implement:
parameters retraction
self-leveling test
macro buttons

**Reply by Thinkersbluff** [reply]
So, when I get better organized, I think issues related to the Klipper component should be managed on the other repo.
For now, this seems to be the best way for me to ask this question.

If you can see PR#9 on the other repo, can you tell me how I am supposed to view the conflicts, so that I can resolve them?
https://github.com/Thinkersbluff/dgus-reloaded_klipper/pull/9

I am getting nowhere trying to use the GitHub CLI on my desktop PC.  I just get error messages from the CLI when I copy/paste the command lines that GitHub is proposing.  150 commits to merge & no idea which one(s) are in conflict or why.

**Reply by henriquevianasilva** [reply]
It seems to me that the problem might just be in the initial readme file.

https://github.com/gbkwiatt/klipper/compare/dgus-reloaded...Thinkersbluff:dgus-reloaded_klipper:DGUS-ReloadedForCR6?diff=unified

**Reply by Thinkersbluff** [reply]
Fwiw, I seem to have successfully updated the modified Klipper on that test branch, in spite of GitHub.  I have not yet committed the update to the default branch but kiauh will happily update your Klipper from the other branch, if you point it there. I just verified that I could use it to build Klipper.bin for STM32 boards and the UI still works correctly.  Let me know if there is a specific way to verify that the Klipper is now up to date, as of yesterday.

**Reply by henriquevianasilva** [reply]
can be consulted here:

Klipper/docs/Releases.md

or you can see the modifications when comparing with original fork.

**Reply by Thinkersbluff** [reply]
Not much different in 0.11.0 that I can test. 
Comparing individual files at random is not a robust check, but it does remind me that Desuuuu also marked-up some of those documents, which my update removed. 🤨

I will delete that branch, then, and try to find out how Desuuuu was doing his updates. It looks like I need to learn a bit about yaml...

**Reply by Thinkersbluff** [reply]
Now that we have posted the klipper.bin files for both the BTT board and the Creality ERA board, we might be able to copy [t5uid1] into the klippy/extras folder of any Klipper installation, to enable this UI.  Not looking to leave behind the 4.5.2 board users, but I need a helper from the Community to validate and share the klipper.bin for 4.5.2 boards.

There is also something known as "wheels" that I would appreciate some help looking into, as another possible way to handle the distribution of [t5uid1].

**Comment by Thinkersbluff** [comment]
At v3.0.2, I added the Nozzle LED On/Off controls to the Miscellaneous Settings menu. (Home->Setup->Miscellaneous Settings)
